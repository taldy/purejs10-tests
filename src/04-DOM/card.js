export function removeCard(id) {
  // do something inmportant
}
