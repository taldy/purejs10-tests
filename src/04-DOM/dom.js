import { removeCard } from './card';

export function insertCardToDOM(card, target) {
  const container = document.createElement('div');
  container.classList.add('card');
  container.setAttribute('data-card-id', card.id);

  const content = document.createElement('p');
  content.append(card.title);
  container.append(content);

  const removeButton = document.createElement('button');
  removeButton.append('X');
  removeButton.addEventListener('click', () => removeCard(card.id));
  container.append(removeButton);

  target.append(container);
}
