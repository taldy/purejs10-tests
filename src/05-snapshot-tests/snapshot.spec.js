import axios from 'axios';
import { insertCardToDOM } from '../04-DOM/dom';

describe('snapshots', () => {
  describe('insertCardToDOM', () => {
    let card;
    let wrapper;

    beforeEach(() => {
      card = {
        id: 111,
        title: 'Learn JS',
        columns: 5,
      };

      wrapper = document.createElement('div');
      document.body.append(wrapper);
    });

    afterEach(() => {
      wrapper.remove();
      wrapper = null;
    });

    it('card html snapshot', () => {
      insertCardToDOM(card, wrapper);
      const cardContainer = wrapper.querySelector('.card');

      expect(cardContainer.outerHTML).toMatchSnapshot();
    });
  });

  it('jsonplaceholder /todos/1 JSON', async () => {
    const res = await axios.get('https://jsonplaceholder.typicode.com/todos/1');
    expect(res.data).toMatchSnapshot();
  });
});
