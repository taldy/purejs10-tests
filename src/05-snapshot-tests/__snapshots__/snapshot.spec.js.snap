// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots insertCardToDOM card html snapshot 1`] = `"<div class=\\"card\\" data-card-id=\\"111\\"><p>Learn JS</p><button>X</button></div>"`;

exports[`snapshots jsonplaceholder /todos/1 JSON 1`] = `
Object {
  "completed": false,
  "id": 1,
  "title": "delectus aut autem",
  "userId": 1,
}
`;
